src/main/kotlin/gropius/model/template/IMSTemplate.kt

```diff
@@ -0,0 +1,32 @@
+package gropius.model.template
+
+import com.expediagroup.graphql.generator.annotations.GraphQLDescription
+import gropius.model.architecture.IMS
+import gropius.model.architecture.IMSIssue
+import gropius.model.architecture.IMSProject
+import io.github.graphglue.model.Direction
+import io.github.graphglue.model.DomainNode
+import io.github.graphglue.model.NodeRelationship
+import org.springframework.data.annotation.Transient
+
+@DomainNode("imsTemplates")
+@GraphQLDescription(
+    """Template for imss
+    Defines templated fields with specific types (defined using JSON schema).
+    """
+)
+class IMSTemplate(
+    name: String, description: String, templateFieldSpecifications: MutableMap<String, String>, isDeprecated: Boolean
+) : Template<IMS, IMSTemplate>(name, description, templateFieldSpecifications, isDeprecated) {
+
+    @NodeRelationship(SubTemplate.PART_OF, Direction.INCOMING)
+    @GraphQLDescription("SubTemplate applied to all IMSProjects with this Template")
+    @delegate:Transient
+    val imsProjectTemplate by NodeProperty<IMSProject>()
```

Comment:
I think this is the wrong type

Reply:
627fa894d0487f997628d7d8f967a37420cce136?